tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -431,72 +431,123 @@ module.exports = {
         this.elements.fileLinkInFileRow.selector
     },
     /**
+     * function checks for the element whether listed or not on the filesList
+     * @param {string} element Name of the file/folder/resource
      *
-     * @param {string} element
+     * @returns {boolean}
      */
-    assertElementNotListed: function (element) {
-      return this
+    isElementListed: async function (element) {
+      let isListed = true
+      await this
         .waitForElementVisible('@filesTable')
         .useXpath()
         .waitForElementNotPresent('@loadingIndicator')
-        .waitForElementNotPresent(this.getFileRowSelectorByFileName(element))
+        .api.elements(
+          'xpath',
+          this.getFileRowSelectorByFileName(element),
+          (result) => {
+            isListed = result.value.length > 0
+          })
+        .useCss()
+      return isListed
     },
     /**
+     * function checks whether sharing button of provided file-row is present
      *
-     * @param {string} element
+     * @param {string} fileName
+     * @returns {Promise<boolean>}
      */
-    assertElementListed: function (element) {
-      return this
-        .waitForElementVisible('@filesTable')
-        .useXpath()
-        .waitForElementNotPresent('@loadingIndicator')
-        .waitForElementPresent(this.getFileRowSelectorByFileName(element))
-    },
-    assertSharingIsDisabled: function (resource) {
-      const resourceRowXpath = this.getFileRowSelectorByFileName(resource)
+    isSharingBtnPresent: async function (fileName) {
+      const resourceRowXpath = this.getFileRowSelectorByFileName(fileName)
       const shareButtonXpath = this.elements.shareButtonInFileRow.selector
       const resourceShareButtonXpath = resourceRowXpath + shareButtonXpath
-      return this
+      let isPresent = true
+      await this
+        .closeSidebar(100)
+        .api.elements(
+          this.elements.shareButtonInFileRow.locateStrategy,
+          resourceShareButtonXpath,
+          (result) => {
+            isPresent = result.value.length > 0
+          })
+      return isPresent
+    },
+    /**
+     * function checks for the provided tabs on the files-page-sidebar are present or not
+     *
+     * @param {string} rowSelector x-path for provided resource row selector
+     * @param {string} tabSelector x-path for provided tab
+     * @returns {Promise<boolean>}
+     */
+    isSidebarTabPresent: async function (rowSelector, tabSelector) {
+      let isPresent = true
+      await this
         .closeSidebar(100)
         .useXpath()
-        .assert.elementNotPresent(resourceShareButtonXpath)
-        .click(resourceRowXpath)
+        .click(rowSelector)
         .waitForElementVisible('@sidebar')
-        .assert.elementNotPresent('@sidebarLinksTab')
-        .assert.elementNotPresent('@sidebarCollaboratorsTab')
+        .api.elements(
+          this.elements.sidebarLinksTab.locateStrategy,
```

Comment:
`tabSelector` comes from outside of the function, so I think we should not rely on `this.elements.sidebarLinksTab.locateStrategy` being the correct strategy.
if `tabSelector` has alsways be an xpath, as mentioned in the comments, you could use `xpath` hardcoded as strategy here

Reply:
OK! :rocket: